### DISCUSSION TITLE
通过Github Action部署后，cf里始终是preview分支

### DISCUSSION THREAD

**Discussion by battlewolf9527** [dreamhunter2333/cloudflare_temp_email#338] [General]
通过Github Action部署后，cf里始终是preview分支

**Comment by dreamhunter2333** [comment]
创建pages 时需要指定生产分支，请指定为 production

**Comment by antergone** [comment]
> 创建pages 时需要指定生产分支，请指定为 production

如果是上传zip包是可以指定的，github action我也没找到指定的地方。我是通过API的方式更新的。
```
curl -X PATCH "https://api.cloudflare.com/client/v4/accounts/{account_id}/pages/projects/{project_name}" \
  -H "Authorization: Bearer <API_TOKEN>" \
  -H "Content-Type: application/json" \
  --data "{\"production_branch\": \"production\"}"
```
文档：https://developers.cloudflare.com/pages/configuration/branch-build-controls/#production-branch-control

如果方案没错的话， @dreamhunter2333 看看是否可以新增到Workflow里

**Comment by dreamhunter2333** [comment]
> > 创建pages 时需要指定生产分支，请指定为 production
> 
> 
> 
> 如果是上传zip包是可以指定的，github action我也没找到指定的地方。我是通过API的方式更新的。
> 
> ```
> 
> curl -X PATCH "https://api.cloudflare.com/client/v4/accounts/{account_id}/pages/projects/{project_name}" \
> 
>   -H "Authorization: Bearer <API_TOKEN>" \
> 
>   -H "Content-Type: application/json" \
> 
>   --data "{\"production_branch\": \"production\"}"
> 
> ```
> 
> 文档：https://developers.cloudflare.com/pages/configuration/branch-build-controls/#production-branch-control
> 
> 
> 
> 如果方案没错的话， @dreamhunter2333 看看是否可以新增到Workflow里
> 
> 
> 
> 
> 
> 

OK 我有空把这个作为一个参数

**Comment by battlewolf9527** [comment]
嗯，我没有上传代码，只是创建了pages，然后就用github action部署了

**Comment by dreamhunter2333** [comment]
> 嗯，我没有上传代码，只是创建了pages，然后就用github action部署了

应该是创建的时候指定的生产分支不是production

**Comment by dreamhunter2333** [comment]
已增加 `FRONTEND_BRANCH`，参考

https://github.com/dreamhunter2333/cloudflare_temp_email/blob/main/vitepress-docs/docs/zh/guide/github-action.md
